sources/Outputs/include/Dyd.h

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  Dyd.h
+ *
+ * @brief Dynaflow launcher DYD file writer header file
+ *
+ */
+
+#pragma once
+
+#include "Algo.h"
+
+#include <DYDBlackBoxModel.h>
+#include <DYDMacroConnect.h>
+#include <DYDMacroConnection.h>
+#include <DYDMacroConnector.h>
+#include <DYDMacroStaticReference.h>
+#include <boost/shared_ptr.hpp>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace dfl {
+namespace outputs {
+
+/**
+ * @brief Dyd writer
+ */
+class Dyd {
+ public:
+  /**
+   * @brief Dyd definition to provide informations to build the Dyd file
+   */
+  struct DydDefinition {
+    /**
+     * @brief Constructor
+     *
+     * @param base the basename for current file (corresponds to filepath basename)
+     * @param filepath the filepath of the dyd file to write
+     * @param gens generators definition coming from algorithms
+     * @param loaddefs load definitions coming from algorithms
+     * @param slacknode the slack node to use
+     */
+    DydDefinition(const std::string& base, const std::string& filepath, const std::vector<algo::GeneratorDefinition>& gens,
+                  const std::vector<algo::LoadDefinition>& loaddefs, const std::shared_ptr<inputs::Node>& slacknode) :
+        basename{base},
+        filename{filepath},
+        generators{gens},
+        loads{loaddefs},
+        slackNode{slacknode} {}
+
+    std::string basename;                               ///< basename for file
+    std::string filename;                               ///< filepath for file to write
+    std::vector<algo::GeneratorDefinition> generators;  ///< generators found
+    std::vector<algo::LoadDefinition> loads;            ///< list of loads
+    std::shared_ptr<inputs::Node> slackNode;            ///< list of generators
```

Comment:
Should it be in the comment the list of generators or the slack node to use ?

Reply:
OK